### ISSUE TITLE
Add tests for using branded types as record keys

### ISSUE THREAD

**Issue by eemelipa** [remeda/remeda#684]
# What?

If branded types are used as a record keys the `mapValues` function does not recognize the value type correctly.

# Reproduce

```ts
import * as R from 'remeda'
import z from 'zod'

const ISODate = z.string().brand('ISODate')
type ISODate = z.infer<typeof ISODate>

const valuesByDate: Record<ISODate, number> = {
  [ISODate.parse('2022-01-01')]: 1,
  [ISODate.parse('2022-01-02')]: 2,
  [ISODate.parse('2022-01-03')]: 3,
}

R.mapValues(valuesByDate, (value) => {
  console.log(value) // value's type is "unknown" when it should be "number"
})
```

**Comment by eemelipa** [comment]
One potential fix for this specific case is to modify in `src/_types.ts` the line

```ts
export type ObjectKeys<T extends object> = `${Exclude<keyof T, symbol>}`;
```

to be

```ts
export type ObjectKeys<T extends object> = Exclude<keyof T, symbol>;
```

That might however have some other side effects

**Comment by cjquines** [comment]
i believe this is fixed on `remeda@beta`, and the change is extensive enough that we probably won't backport it, so i'm gonna close this for now

**Comment by eranhirsch** [comment]
We can keep this task to add a test for it; it's good that we track that we don't break such use-cases.

**Comment by cjquines** [comment]
closed by #707
